### ISSUE TITLE
Cannot open encrypted library - always get error message about wrong password

### ISSUE THREAD

**Issue by m-andre** [haiwen/seafile-iOS#87]
I cannot open the library with password - I always get the error message about wrong password library - even though I can open it in web interface or in desktop client with the same password and I triple-checked I enter the password correctly.

I am running:
* Seafile Pro Server (Free) version 4.3.3
* iOS Seafile Pro Client version 2.1.1

**Comment by m-andre** [comment]
I noticed that everytime I try to access encrypted library from the iOS client, I get this in web server access log - note HTTP return code 440:
`"POST /api2/repos/<repo-id-removed>/?op=setpassword HTTP/1.1" 440 1045 "-" "SeafilePro/1 CFNetwork/711.5.6 Darwin/14.0.0"`

Am I missing some configuration or is this not actually supported on iOS?

**Comment by poetwang** [comment]
Can it work on seacloud.cc?

**Comment by m-andre** [comment]
Yes, it works on seacloud.cc - is it something in my configuration?

**Comment by m-andre** [comment]
Alright, I did a bit of sniffing and now I know what is going on:
My password starts with semicolon (;) - apparently this messes up iOS client (on my installation as well as on seacloud.cc). Webclient and desktop client do not mind passwords starting with semicolon.
Would it be possible to enable also these kind of passwords in iOS client?

On the side-note:
I've also noticed that web-interface does not make the same call as iOS client: 
iOS client calls `/api2/repos/<repo-id>/?op=setpassword`, while webclient calls `/repo/set_password/` and sends as a POST parameters `repo_id`, `username` and `password` (in cleartext!).
